### PR TITLE
[AMD-AIE] Add support for vector.contract in insert-cores pass

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
-- This commit adds suppport for vector.contract in `--iree-amdaie-insert-cores`
   pass.

Co-authored-by: Abhishek Varma <abhvarma@amd.com>
Co-authored-by: Vivian Zhang <vivian.zhang@amd.com>

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp [hunk 1]
```diff
@@ -111,6 +112,16 @@ LogicalResult insertCoreOps(mlir::ModuleOp moduleOp) {
       } else if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
         rewriter.setInsertionPoint(endOp);
         rewriter.moveOpBefore(linalgOp, endOp);
+      } else if (isa<vector::ContractionOp>(op)) {
+        // Because vector.contract op is surrounded by vectorized loop nest, we
+        // need to traverse to the outermost loop to move the entire vectorized
+        // computation within amdaie.core op.
+        Operation *currOp = op;
+        while (currOp->getParentOp() != forallOp) {
+          currOp = currOp->getParentOp();
```
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp:121]: I think you could just use currOp = forallOp? Also does it make sense to even continue the walk if the whole loop is already moved inside the amdaie.core op?

Comment (nirvedhmeshram) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp:121]: actually looking for at the code, I see you want and an op whose parent is the forallOp, but from the IR it looks like the consequence is the whole loop nest gets moved, so what I was suggesting would work too?

**Review by yzhang93 — COMMENTED** [review]
Thanks! LGTM. I'd like @jtuyls to review this.

**Review by jtuyls — APPROVED** [review]
LGTM

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp (hunk 1 above)
Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp:121]: Discussed offline. Marking resolved.

**Pull request merged**